scripts/verify_dataset.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+# coding: utf8
+'''Compute various data integrity checks for the OpenMIC dataset.
+
+Example
+-------
+$ ./scripts/verify_dataset.py path/to/openmic path/to/checksums
+'''
+
+import argparse
+import datetime
+import glob
+from joblib import Parallel, delayed
+import json
+import numpy as np
+import os
+import pandas as pd
+import soundfile as sf
+import sys
+import warnings
+
+import openmic.util
+
+
+def hash_one(fname):
+    fkey = openmic.util.filebase(fname)
+    fhash = openmic.util.md5_file(fname)
+    return (fkey, fhash)
+
+
+def hash_collection(filepaths, n_jobs, verbose):
+    dfx = delayed(hash_one)
+    pool = Parallel(n_jobs=n_jobs, verbose=verbose)
+
+    return pd.DataFrame.from_dict(dict(pool(dfx(fn) for fn in filepaths)),
+                                  columns=['md5'], orient='index')
+
+
+def check_md5(filepaths, checksum_file, n_jobs, verbose):
+    df = hash_collection(filepaths, n_jobs, verbose)
+
+    exp = pd.read_csv(checksum_file, index_col=0)
+    df = df.join(exp, how='outer', rsuffix='_expected')
+    if len(df) != len(filepaths):
+        raise warnings.warn('File key mismatch! {} total rows, expected {}'
+                            .format(len(df), len(filepaths)))
+
+    success = (df.md5 == df.md5_expected).all()
+    if not success:
+        percent = 100 * (df.md5 != df.md5_expected).mean()
+        raise warnings.warn('MD5 hash mismatch on {:0.2f}% of records'
+                            .format(percent))
+
+    return success
+
+
+def _check_duration(fname, expected_duration, tolerance):
+    dur = sf.info(fname).duration
```

Comment:
Just a heads-up on this one: `sf.info()` uses header data to infer duration, and this isn't always reliable.  See https://github.com/librosa/librosa/issues/686 .

I think it's fine to leave it as is here, but maybe put a comment in place that this isn't not 100% foolproof.

Reply:
gah, dammit ... this is the same problem that sox bit me on the bogus [FMA headers](https://github.com/mdeff/fma/issues/27). 

I'd rather load the audio to compute duration post-decoding and eat that computational cost, since this is supposed to be an infrequent but robust guarantee that things are legit. MD5 checksums would catch deltas, but this would help identify this specific (known) issue.